src/journal/logic.py

```diff
@@ -325,6 +326,36 @@ def unset_article_session_variables(request):
     return redirect("{0}?page={1}".format(reverse('journal_articles'), page))
 
 
+def handle_search_controls(request):
+    if request.POST:
+        search_term = request.POST.get('article_search', False)
+        keyword = request.POST.get('keyword', False)
+        sort = request.POST.get('sort', False)
+            
+        return search_term, keyword, sort, set_search_GET_variables(request, search_term, keyword, sort)
+
+    else:
+        search_term = request.GET.get('article_search', False)
+        keyword = request.GET.get('keyword', False)
+        sort = request.GET.get('sort', False)
+        
+                
+        return search_term, keyword, sort, None
+
+def set_search_GET_variables(request, search_term, keyword, sort):
```

Comment:
A few suggestions on this function:
 - If the arguments are optional, the signature should reflect that: `def set_search_GET_variables(request, search_term=None, keyword=None, sort=None)`
- `redir_str` will be undefined if neither of `search_term`, `keyword` or `sort` are truthy